tests/ui/infinite/infinite-instantiation-struct-tail-ice-114484.rs

```diff
@@ -9,6 +9,7 @@
 // Regression test for #114484: This used to ICE during monomorphization, because we treated
 // `<VirtualWrapper<...> as Pointee>::Metadata` as a rigid projection after reaching the recursion
 // limit when finding the struct tail.
+#![allow(returning_pointers_to_local_variables)]
```

Comment:
Which part of the test was triggering the lint? All the returned poiunters in this test look legit to me.

Reply:
You are right! The problem was the `gen_vtable_ptr` which returns `*const ()`, which should be fine. I'll add an exclude for the unit type :)

Reply:
I don't think excluding the unit type is correct: it's not particularly special. Where in that test did your lint think there was a pointer to stack memory? The `vtable` in `gen_vtable_ptr` was transmuted from pointer metadata which is not pointing to stack memory, and the `ptr` in `virtualize_my_trait` is obtained by transmuting an existing pointer.

Shouldn't your lint be operating on MIR actually? Because right now operating on HIR you have to think about implicit coercions and reborrows and things.